src/aiohabit/outputs/ansible_inventory.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ansible inventory output module."""
+
+from copy import deepcopy
+from aiohabit.outputs.utils import is_windows_host, resolve_hostname
+from aiohabit.utils import get_config_value, get_host_from_metadata, save_yaml
+
+DEFAULT_INVENTORY_PATH = "aiohabit-inventory.yaml"
+DEFAULT_INVENTORY_LAYOUT = {"all": {"children": [], "hosts": {}}}
+
+
+def copy_meta_attrs(host, meta_host, attrs):
+    """Copy attributes from host metadata entry and prefixing them with meta."""
+    for attr in attrs:
+        val = meta_host.get(attr)
+        if val:
+            host[f"meta_{attr}"] = val
+
+
+def find_group(inventory, group_names, top=True):
+    """Find first matching group in inventory which has one of the provided names."""
+    if not inventory:
+        return None
+
+    groups = inventory.keys()
+    for group in groups:
+        if group in group_names:
+            return inventory[group]
+
+    for group in groups:
+        g = inventory[group]
+
+        if "children" not in g:
+            continue
+
+        found = find_group(g["children"], group_names, top=False)
+        if found:
+            return found
+
+    if top:
+        return inventory["all"]
+
+    return None
+
+
+class AnsibleInventoryOutput:
+    """
+    Generate Ansible inventory with provisioned machines.
+
+    The resulting inventory combines data from provisioning config, job metadata
+    files and information in DB.
+    """
+
+    def __init__(self, config, db, metadata, path=DEFAULT_INVENTORY_PATH):
+        """Init the output module."""
+        self._config = config
+        self._db = db
+        self._metadata = metadata
+        self._path = path
+
+    def create_ansible_host(self, name):
+        """Create host entry for Ansible inventory."""
+        meta_host, meta_domain = get_host_from_metadata(self._metadata, name)
+        db_host = self._db.hosts[name]
+
+        ip = db_host.ips[0]
+        ansible_host = resolve_hostname(ip) or ip
+
+        python = self._config["python"] or meta_host["os"]
```

Comment:
First of all, thanks for the work done!
When trying the generation i bumped into this line in utput:
```yaml
ansible_python_interpreter: *id001
```
which may be because the 
```python
self._config["python"] is a dictionary
```
it is probably some minor bug or the difference in provisioning config which i use:
```yaml
# ansible_python_interpreter
python:
    fedora-28: /usr/bin/python3
    fedora-29: /usr/bin/python3
    fedora-30: /usr/bin/python3
    fedora-31: /usr/bin/python3
    fedora-32: /usr/bin/python3
    fedora-rawhide: /usr/bin/python3
    rhel-6.10: /usr/bin/python
    rhel-7.0: /usr/bin/python
    rhel-7.1: /usr/bin/python
    rhel-7.2: /usr/bin/python
    rhel-7.3: /usr/bin/python
    rhel-7.4: /usr/bin/python
    rhel-7.5: /usr/bin/python
    rhel-7.5-atomic: /usr/bin/python
    rhel-7.6-atomic: /usr/bin/python
    default: /usr/libexec/platform-python
```

Reply:
This worked for me:
```suggestion
        python = self._config["python"][meta_host["os"]] or self._config["python"]["default"]
```
could you please check if that match the intention?

result
```yaml
      ansible_python_interpreter: /usr/bin/python3
```